sidekick_core/lib/src/version_checker.dart

```diff
@@ -5,32 +5,45 @@ import 'package:pub_semver/pub_semver.dart';
 import 'package:sidekick_core/sidekick_core.dart';
 import 'package:yaml/yaml.dart';
 
-/// Checks and updates dependencies of the given [package]
+/// Checks and updates dependencies
 class VersionChecker {
-  const VersionChecker(this.package);
+  const VersionChecker();
 
-  final DartPackage package;
+  /// Checks whether the latest version of [package] is [version]
+  Future<bool> isPackageUpToDate({
```

Comment:
`VersionChecker` now has no properties. You could either make all methods `static` or remove the class altogether, leaving only top-level functions.

It could make testing harder, but eventually not, that should be explored

Reply:
I decided to make all methods `static` and the class abstract because I didn't want to lose IDE autocompletion when typing `VersionChecker.` as these functions are related